addons/mail/static/src/models/message/message.js

```diff
@@ -116,15 +72,34 @@ function factory(dependencies) {
                     });
                     channelList.push(channel);
                 }
-                data2.threadCaches.push(['replace', channelList.map(channel => channel.mainCache)]);
+                data2._threads = [
+                    ['unlink-filtered', thread => thread.model === 'mail.channel'],
+                    ['link', channelList],
```

Comment:
From https://github.com/odoo/odoo/pull/53297#discussion_r442986100 I think it's no longer useful to have `unlink-filtered`. That's certainly an non-necessary command, I had to introduce because field `_threads` mixes too many things together and the solution to this problem is to split in many different fields, no have more magic commands.